diff/walking/differ.go

```diff
@@ -69,13 +70,19 @@ func (s *walkingDiff) Compare(ctx context.Context, lower, upper []mount.Mount, o
 		config.MediaType = ocispec.MediaTypeImageLayerGzip
 	}
 
-	var isCompressed bool
-	switch config.MediaType {
-	case ocispec.MediaTypeImageLayer:
-	case ocispec.MediaTypeImageLayerGzip:
-		isCompressed = true
+	dc, err := images.DiffCompression(ctx, config.MediaType)
+	if err != nil {
+		return emptyDesc, errors.Wrapf(err, "unsupported diff media type: %v", config.MediaType)
+	}
+
+	var alg compression.Compression
+	switch dc {
+	case "gzip":
+		alg = compression.Gzip
```

Comment:
Wouldn't it be better to return `compression.XYZ` directly from `DiffCompression` to avoid extra switch? `compression` package expects `compression.Compression` type anyway.

Reply:
If we were to return that, it would be better to move it out of the `images` package to limit its dependency graph. Right now they are intended to be different though as there may be more compression types from the media types than supported in the compression package. Likewise handling the "unknown" logic is kind of weird and would prefer to handle it outside the compression package. What might make more sense though is to have a package just for handling the media types.